flink-table/flink-sql-client/src/main/java/org/apache/flink/table/client/gateway/local/LocalExecutor.java

```diff
@@ -281,13 +283,18 @@ public void setSessionProperty(String sessionId, String key, String value) throw
 		ExecutionContext<?> context = getExecutionContext(sessionId);
 		Environment env = context.getEnvironment();
 		Environment newEnv = Environment.enrich(env, ImmutableMap.of(key, value), ImmutableMap.of());
+		ExecutionContext.SessionState sessionState = context.getSessionState();
+		// update table config
+		newEnv.getConfiguration().asMap().forEach((k, v) ->
+				sessionState.config.getConfiguration().setString(k, v));
```

Comment:
Will this influence the `RESET` command which should clear all session properties?

Reply:
good catch, we should store the original `SessionState` for resetting all properties to initial state.